jina/docker/hubio.py

```diff
@@ -73,7 +74,46 @@ def new(self):
             self.logger.info('nothing is created, bye!')
 
     def push(self, name: str = None, readme_path: str = None):
-        """A wrapper of docker push """
+        """ A wrapper of docker push """
+        is_build_success, is_push_success = True, False
+        try:
+            # self._push(name, readme_path)
+            is_push_success = True
+        except Exception:
+            self.logger.error(f'can not push to the registry')
+        
+        file_path = f'/tmp/{self.args.name.replace("/", "_")}.json'
+        if os.path.isfile(file_path):
+            import json
+            with open(file_path) as f:
+                result = json.load(f)
+        
+        else:
+            image = self._client.images.get(name)
+            _details = {
+                'inspect': self._raw_client.inspect_image(image.tags[0]),
+                'tag': image.tags[0],
+                'hash': image.short_id,
+                'size': get_readable_size(image.attrs['Size']),
+            }
+            
+            result = {
+                'name': name,
+                'path': getattr(self.args, 'path', ''),
+                'details': _details,
+                'last_build_time': get_now_timestamp(),
+                'build_duration': '',
+                'is_build_success': is_build_success,
+                'is_push_success': is_push_success,
+                'build_logs': [],
```

Comment:
In case build logs are not found in the temporary location - does it mean that the build was never triggered or it was an unsuccessful build? 
In the first case, we'd have to trigger the build right?
Also, `build_duration` is an empty field. In this case `is_build_success` and `is_push_success` can never be true (it makes sense to set it false here?)

Reply:
In `push` logic, we have a `_check_docker_image` method, which checks if the image was built locally or not. 

For the case where the tmp file doesn't exist 

- The user might not have built the image. During `_push`, we'll check if it is a valid image in the local machine. If `_push` raises an exception, we'll never write to the db

- Even though the build was successful, the user might have deleted the file manually or for some reason, the file didn't get created. In this case, due to lack of build information, we'll write `build_duration` as empty, `is_build_success` is assumed to be True, `is_push_success` depends on `_push` call.

Reply:
From L91 to 107, I think we don't need it. This surpasses our (implicit) validation in the build process, and allows user to upload meta info of an **arbitrary image**, which is not preferred.